### ISSUE TITLE
Version 4.40.4 of YQ does not parse "----" input as expected

### ISSUE THREAD

**Issue by pranay-harness** [mikefarah/yq#1890]
**Describe the bug**
Version 4.40.4 of YQ does not parse "----" input as expected. It was working perfectly fine with previous version of YQ.

**Version of yq: 4.40.4**
Operating system: linux
Installed via: binary release

**Input Yaml**

config1.yml:
```yaml
server:
backstageSaToken:
```

**CLI command that does not work as expected:**
```CONFIG_KEY="your_key" CONFIG_VALUE="----BEGIN" yq -i eval '.[$CONFIG_KEY] = env(CONFIG_VALUE)' config1.yml" config1.yml```

**Output:**
```cat config1.yml
server:
backstageSaToken:
```

**Version of yq: 4.35.2**
Operating system: linux
Installed via: binary release

config1.yml:
```yaml
server:
backstageSaToken:
```

**CLI command that works as expected:**
```CONFIG_KEY="your_key" CONFIG_VALUE="----BEGIN" yq -i eval '.[$CONFIG_KEY] = env(CONFIG_VALUE)' config1.yml```

**Output:**
```cat config.yml
server: '----BEGIN'
backstageSaToken: '----BEGIN'
```

**Additional context**
The CLI command in both cases remains the same.

**Comment by mikefarah** [comment]
The `env` command parses the given input (and so you can give it a number/boolean/array and it will be the correct type. `---` is a document separator in yaml, and that's what it is being interpreted as.

To force it to parse as a string, use `strenv` instead, and you'll get what you need.

**Comment by pranay-harness** [comment]
So what got changed between 4.35.2 and 4.40.4? It was perfectly working fine in the previous version, was there any change from your side?

**Comment by bharatgoelharness** [comment]
@mikefarah do you got a chance to look into this?

**Comment by mikefarah** [comment]
That's a good point re version change, not sure what would have caused that. Haven't had a chance to look at it sorry - but I'll get to it.

**Comment by mikefarah** [comment]
Found the issue :) will be fixed in the next release

**Comment by mikefarah** [comment]
Fixed in 4.40.5
